engine/src/main/java/org/hibernate/validator/internal/engine/constraintvalidation/ConstraintTree.java

```diff
@@ -148,9 +151,9 @@ private ConstraintTree(ConstraintDescriptorImpl<A> descriptor, Type validatedVal
 		}
 	}
 
-	private void throwExceptionForNullValidator(Type validatedValueType, String path) {
+	private ValidationException getExceptionForNullValidator(Type validatedValueType, String path) {
 		if ( descriptor.getConstraintType() == ConstraintDescriptorImpl.ConstraintType.CROSS_PARAMETER ) {
-			throw log.getValidatorForCrossParameterConstraintMustEitherValidateObjectOrObjectArrayException(
```

Comment:
What's the motivation for throwing it separately instead doing it here? I think stack traces are a tiny bit confusing in this case due to line number mismatches, so I'd only do it if there's strong advantage.

Reply:
At one point I needed it because I used `Optional.orElseThrow` but it's not strictly needed anymore.

That being said, I kept it because I like when the compiler is able to follow the flow and if we do the throw in another method it is not. So, I'd really prefer to keep it this way.

Reply:
Ok.